one-ui/src/main/java/com/marklogic/hub/curation/services/FlowManagerService.java

```diff
@@ -0,0 +1,288 @@
+package com.marklogic.hub.curation.services;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.FlowManager;
+import com.marklogic.hub.StepDefinitionManager;
+import com.marklogic.hub.error.DataHubProjectException;
+import com.marklogic.hub.oneui.exceptions.DataHubException;
+import com.marklogic.hub.flow.Flow;
+import com.marklogic.hub.flow.impl.FlowImpl;
+import com.marklogic.hub.oneui.models.StepModel;
+import com.marklogic.hub.scaffold.Scaffolding;
+import com.marklogic.hub.step.StepDefinition;
+import com.marklogic.hub.step.impl.Step;
+import com.marklogic.hub.util.json.JSONObject;
+import com.marklogic.hub.util.json.JSONUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+@Service
+public class FlowManagerService {
+
+    @Autowired
+    private FlowManager flowManager;
+
+    @Autowired
+    private StepDefinitionManager stepDefinitionManager;
+
+    @Autowired
+    private Scaffolding scaffolding;
+
+    public Flow createFlow(String flowJson, boolean isFlowPresent) {
+        JSONObject jsonObject;
+        try {
+            jsonObject = new JSONObject(flowJson);
+
+            JSONUtils.trimText(jsonObject, "separator");
+        }
+        catch (IOException e) {
+            throw new DataHubException("Unable to parse flow json string : " + e.getMessage());
+        }
+
+        if (!jsonObject.isExist("name") || StringUtils.isEmpty(jsonObject.getString("name"))) {
+            throw new BadRequestException("Flow Name not provided. Flow Name is required.");
+        }
+
+        String flowName = jsonObject.getString("name");
+        Flow flow;
+        if (!isFlowPresent) {
```

Comment:
This all seems logic that would go into the DS endpoint. @aebadirad What are your thoughts on that? That would require calling ML first. But it would get rid of a ton of code in Java and move it into a more interesting DS endpoint that can see if the flow exists already, and create it if not or update it otherwise.